src/main/java/com/github/splendor_mobile_game/database/InMemoryDatabase.java

```diff
@@ -109,6 +149,12 @@ public void loadCards() {
         }
     }
 
+    @Override
+    public ArrayList<Noble> getAllNobles(){
+        ArrayList<Noble> array=new ArrayList<>(this.allNobles);
```

Comment:
`return this.allNobles` will be a sufficient sollution

Reply:
I think we should return the copy, because what if by accident someone change this array when the get it? `allNobles` should be immutable by design, so returning a copy is good design enforcement.

Reply:
Oh, that's right. Didn't think of that. Let's use the old solution then.